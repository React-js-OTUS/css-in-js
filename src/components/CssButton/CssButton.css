.CssButton__button {
  border-radius: 8px;
  font-size: 18px;
  padding: 8px 16px;
  outline: none;
  border: 0;
  background-color: #00CC00;
  color: #fff;
  cursor: pointer;
}

.CssButton__button:hover {
  background-color: #39E639;
}

.CssButton__button:active {
  background-color: #269926;
}

.CssButton__text {
  font-size: 18px;
}

.CssButton__text_opened {
  display: block;
}

.CssButton__text_closed {
  display: none;
}
