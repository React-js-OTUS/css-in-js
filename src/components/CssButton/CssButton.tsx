import React, { FC, useState } from 'react';
import cn from 'clsx';
import './CssButton.css';

export type ButtonProps = {
  className?: string;
};

export const CssButton: FC<ButtonProps> = ({ className, ...props }) => {
  const [opened, setOpened] = useState<boolean>(false);
  return (
    <div className={cn('CssButton', className)}>
      <button className={'CssButton__button'} {...props} onClick={() => setOpened(v => !v)} />
      <div className={cn('CssButton__text', opened ? 'CssButton__text_opened' : 'CssButton__text_closed')}>Test</div>
      <div className={cn('CssButton__text', !opened && 'CssButton__text_closed')}>Test</div>
    </div>
  );
}
