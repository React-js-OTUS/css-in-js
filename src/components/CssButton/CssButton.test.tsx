import React from 'react';
import { render } from '@testing-library/react';
import { CssButton } from './CssButton';

describe('CssButton', () => {
  it('base snapshot', () => {
    const { container } = render(<CssButton>CssButton</CssButton>);
    expect(container).toMatchSnapshot();
  });
});
