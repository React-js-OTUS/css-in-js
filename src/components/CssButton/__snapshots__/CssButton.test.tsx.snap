// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CssButton base snapshot 1`] = `
<div>
  <div
    class="CssButton"
  >
    <button
      class="CssButton__button"
    >
      CssButton
    </button>
    <div
      class="CssButton__text CssButton__text_closed"
    >
      Test
    </div>
    <div
      class="CssButton__text CssButton__text_closed"
    >
      Test
    </div>
  </div>
</div>
`;
